base/range.jl

```diff
@@ -415,6 +415,13 @@ end
 
 function getindex{T}(v::Range{T}, i::Integer)
     @_inline_meta
+    ret = convert(T, first(v) + i - 1)
+    @boundscheck ((i > 0) & (ret <= v.stop)) || throw_boundserror(v, i)
```

Comment:
Thanks for looking into that! I was thinking of similar silly edge case where `i = typemax(Int)`.

It's interesting that using `a < b < c` lowers to the short-circuiting version `(a < b) && (b < c)`.